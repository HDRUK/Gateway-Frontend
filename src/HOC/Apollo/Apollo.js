import React from 'react';
import ApolloClient from 'apollo-boost';
import { ApolloProvider } from 'react-apollo';

const client = new ApolloClient({ uri: process.env.REACT_APP_GRAPH_QL_ENDPOINT });

const ApolloApp = props => (
  <ApolloProvider client={client}>
    {props.children}
  </ApolloProvider>
);

export default ApolloApp;