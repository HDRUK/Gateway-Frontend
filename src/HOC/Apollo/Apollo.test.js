import React from "react";
import { create } from "react-test-renderer";
import Apollo from "./Apollo";

describe("<Apollo> ", () => {
    let renderedComponent;

    const envVarMock = {
        REACT_APP_GRAPH_QL_ENDPOINT: "REACT_APP_GRAPH_QL_ENDPOINT"
    };

    beforeAll(() => {
        process.env.REACT_APP_GRAPH_QL_ENDPOINT = envVarMock.REACT_APP_GRAPH_QL_ENDPOINT;
    });

    beforeEach(() => {
        renderedComponent = create(
            <Apollo>
                <div>test</div>
            </Apollo>
        );
    });

    it("should test the Apollo HOC", () => {
        expect(process.env.REACT_APP_GRAPH_QL_ENDPOINT).toEqual(envVarMock.REACT_APP_GRAPH_QL_ENDPOINT);
        expect(renderedComponent).toMatchSnapshot();
    });
});
