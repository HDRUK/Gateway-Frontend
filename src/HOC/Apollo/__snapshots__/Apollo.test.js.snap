// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Apollo>  should test the Apollo HOC 1`] = `
<div>
  test
</div>
`;
